reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/tools/ToolSelection.kt

```diff
@@ -3,45 +3,112 @@ package com.xebia.functional.xef.reasoning.tools
 import com.xebia.functional.xef.auto.CoreAIScope
 import com.xebia.functional.xef.llm.ChatWithFunctions
 import com.xebia.functional.xef.prompt.experts.ExpertSystem
-import com.xebia.functional.xef.reasoning.internals.callModel
 import io.github.oshai.kotlinlogging.KotlinLogging
+import love.forte.plugin.suspendtrans.annotation.JvmAsync
+import love.forte.plugin.suspendtrans.annotation.JvmBlocking
 
 class ToolSelection(
   private val model: ChatWithFunctions,
   private val scope: CoreAIScope,
+  private val tools: List<Tool>,
   private val instructions: List<String> = emptyList()
-) {
+) : Tool {
 
   private val logger = KotlinLogging.logger {}
 
-  suspend fun selectTool(task: String, tools: List<Tool>): ToolSelectionResult {
-    logger.info { "🔍 Selecting tool for task: $task" }
-    return callModel<ToolSelectionResult>(
-        model,
-        scope,
-        prompt =
-          ExpertSystem(
-            system =
-              "You are an expert in tool selection that can choose the best tool for a specific task based on the tool descriptions",
-            query =
-              """|
+  override val name: String = "Tool Selection"
+
+  override val description: String = "Select the best tool for the job"
+
+  override suspend fun invoke(input: String): String {
+    return when (val trace = applyInferredTools(input)) {
+      is ToolsExecutionTrace.Completed -> trace.output
+      ToolsExecutionTrace.Empty -> ""
+    }
+  }
+
+  @JvmBlocking
+  @JvmAsync
```

Comment:
@nomisRev This seems to do similar to what we were attempting to build with `@AsFuture` .
This creates an async and sync method and makes the suspend method @JvmSynthetic. https://github.com/ForteScarlet/kotlin-suspend-transform-compiler-plugin